usaspending_api/disaster/v2/views/disaster_base.py

```diff
@@ -24,46 +24,52 @@ def covid_def_code_strings():
     return [code["code"] for code in covid_def_codes().values("code")]
 
 
-def latest_gtas_of_each_year_queryset():
+def latest_gtas_of_each_year_queryset() -> GTASSF133Balances:
     q = Q()
-    for final_for_fy in finals_for_fy(False):
-        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[1])
-    for final_for_fy in finals_for_fy(True):
-        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[1])
+    for sub in final_submissions_for_all_fy():
+        if not sub.is_quarter:
+            q |= Q(fiscal_year=sub.fiscal_year) & Q(fiscal_period=sub.fiscal_period)
     if not q:
         return GTASSF133Balances.objects.none()
     return GTASSF133Balances.objects.filter(q)
 
 
-def latest_faba_of_each_year_queryset():
-    q = Q()
-    for final_for_fy in finals_for_fy(False):
-        q |= Q(submission__reporting_fiscal_year=final_for_fy[0]) & Q(
-            submission__reporting_fiscal_period=final_for_fy[1]
-        )
-    for final_for_fy in finals_for_fy(True):
-        q |= Q(submission__reporting_fiscal_year=final_for_fy[0]) & Q(
-            submission__reporting_fiscal_period=final_for_fy[1]
-        )
+def latest_faba_of_each_year_queryset() -> FinancialAccountsByAwards:
+    q = filter_by_latest_closed_periods()
     if not q:
         return FinancialAccountsByAwards.objects.none()
     return FinancialAccountsByAwards.objects.filter(q)
 
 
-def finals_for_fy(is_quarter):
-    return (
-        DABSSubmissionWindowSchedule.objects.filter(
-            submission_reveal_date__lte=datetime.now(timezone.utc), is_quarter=is_quarter
+def filter_by_latest_closed_periods() -> Q:
+    """Return Django Q for all latest closed submissions (quarterly and monthly)"""
+    q = Q()
+    for sub in final_submissions_for_all_fy():
+        q |= (
+            Q(submission__reporting_fiscal_year=sub.fiscal_year)
+            & Q(submission__quarter_format_flag=sub.is_quarter)
+            & Q(submission__reporting_fiscal_period=sub.fiscal_period)
         )
-        .values("submission_fiscal_year")
-        .annotate(max_submission_fiscal_month=Max("submission_fiscal_month"))
-        .values_list("submission_fiscal_year", "max_submission_fiscal_month")
+    return q
+
+
+def final_submissions_for_all_fy() -> List[tuple]:
+    """
+        Returns a list the latest monthly and quarterly submission for each
+        fiscal year IF it is "closed" aka ready for display on USAspending.gov
+    """
+    return (
+        DABSSubmissionWindowSchedule.objects.filter(submission_reveal_date__lte=datetime.now(timezone.utc))
+        .values("submission_fiscal_year", "is_quarter")
+        .annotate(fiscal_year=F("submission_fiscal_year"), fiscal_period=Max("submission_fiscal_month"))
+        .values_list("fiscal_year", "is_quarter", "fiscal_period", named=True)
```

Comment:
I thought this worked by using the field names of the model class you were querying, but I guess it makes sense that you could and sometimes need to alias those fields. Looks great though and makes the rest of the code very readable.

Reply:
(Sorry I missed this comment on Thursday) 
@kwhickey it would work to keep the column names (except `submission_fiscal_month` since that is an annotation) but I thought abstracting the names would be clearer